rosbridge_server/scripts/rosbridge_websocket.py

```diff
@@ -70,56 +74,49 @@ def open(self):
                          str(exc))
         rospy.loginfo("Client connected.  %d clients total.",
                       clients_connected)
-        if authenticate or enable_authentication:
+        if authenticate_mac or authenticate_user_id_password:
             rospy.loginfo("Awaiting proper authentication...")
 
     def on_message(self, message):
-        global authenticate, enable_authentication
+        global authenticate_mac, authenticate_user_id_password
         # check if we need to authenticate
 
-        if authenticate and not self.authenticated:
+        if authenticate_mac or authenticate_user_id_password \
+                and not self.authenticated:
             try:
                 msg = json.loads(message)
                 if msg['op'] == 'auth':
-                    # check the authorization information
-                    auth_srv = rospy.ServiceProxy('authenticate',
-                                                  Authentication)
-                    resp = auth_srv(msg['mac'], msg['client'], msg['dest'],
-                                    msg['rand'], rospy.Time(msg['t']),
-                                    msg['level'], rospy.Time(msg['end']))
-                    self.authenticated = resp.authenticated
+                    # check what type of authorithation is required and
+                    # and if is is enabled
+                    if msg['method'] == 'mac' \
```

Comment:
can we add an *and* clause here for the case where the 'method' value is missing, to support backwards compatibility?